test/workflows_test.go

```diff
@@ -0,0 +1,1087 @@
+				Name:   "Test",
+				Active: true,
+				Conditions: []conditions.ConditionsRequest{
+					getEntityCondition(),
+					getEventCondition(),
+					getProcessingChannelCondition(),
+				},
+				Actions: []actions.ActionsRequest{
+					getWebhookAction(),
+				},
+			},
+			checker: func(response *common.IdResponse, err error) {
+				assert.Nil(t, err)
+				assert.NotNil(t, response)
+				assert.Equal(t, http.StatusCreated, response.HttpMetadata.StatusCode)
+				assert.NotNil(t, response.Id)
+			},
+		},
+		{
+			name: "when conditions missing then should return error",
+			request: workflows.CreateWorkflowRequest{
+				Name:   "Test",
+				Active: true,
+				Actions: []actions.ActionsRequest{
+					getWebhookAction(),
+				},
+			},
+			checker: func(response *common.IdResponse, err error) {
+				assert.Nil(t, response)
+				assert.NotNil(t, err)
+				chkErr := err.(errors.CheckoutAPIError)
+				assert.Equal(t, http.StatusUnprocessableEntity, chkErr.StatusCode)
+				assert.Contains(t, chkErr.Data.ErrorCodes, "conditions_required")
+			},
+		},
+		{
+			name: "when actions missing then should return error",
+			request: workflows.CreateWorkflowRequest{
+				Name:   "Test",
+				Active: true,
+				Conditions: []conditions.ConditionsRequest{
+					getEntityCondition(),
+					getEventCondition(),
+					getProcessingChannelCondition(),
+				},
+			},
+			checker: func(response *common.IdResponse, err error) {
+				assert.Nil(t, response)
+				assert.NotNil(t, err)
+				chkErr := err.(errors.CheckoutAPIError)
+				assert.Equal(t, http.StatusUnprocessableEntity, chkErr.StatusCode)
+				assert.Contains(t, chkErr.Data.ErrorCodes, "actions_required")
+			},
+		},
+	}
+
+	client := DefaultApi().WorkFlows
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			response, err := client.CreateWorkflow(tc.request)
+
+			tc.checker(response, err)
+
+			if response != nil {
+				if _, err := client.RemoveWorkflow(response.Id); err != nil {
+					assert.Fail(t, fmt.Sprintf("Failed to remove workflow: %s", err.Error()))
+				}
+			}
+		})
+	}
+}
+
+func TestGetWorkflows(t *testing.T) {
+	workflow := createWorkflow(t)
+
+	cases := []struct {
+		name    string
+		checker func(*workflows.GetWorkflowsResponse, error)
+	}{
+		{
+			name: "when retrieving all workflows then it returns existing workflows",
+			checker: func(response *workflows.GetWorkflowsResponse, err error) {
+				assert.Nil(t, err)
+				assert.NotNil(t, response)
+				assert.NotNil(t, response.Workflows)
+				assert.NotEmpty(t, response.Workflows)
+				for _, workflow := range response.Workflows {
+					assert.NotNil(t, workflow.Id)
+					assert.NotNil(t, workflow.Name)
+					assert.True(t, workflow.Active)
+				}
+			},
+		},
+	}
+
+	client := DefaultApi().WorkFlows
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.checker(client.GetWorkflows())
+		})
+	}
+
+	if _, err := client.RemoveWorkflow(workflow.Id); err != nil {
```

Comment:
There are a few places that you are using the same evaluation and same logic, which makes me think that you can create a function like
`func removeWorkflow(t *testing.T, client *workflows.Client, workflow *common.IdResponse) {
	if _, err := client.RemoveWorkflow(workflow.Id); err != nil {
		assert.Fail(t, fmt.Sprintf("Failed to remove workflow: %s", err.Error()))
	}
}`

and call it 
`removeWorkflow(t, client, workflow)`

or even better, use the `t.Cleanup`
so that it doesn't matter if you have an error or not, you will always delete the created webhook